event/subscription_test.go

```diff
@@ -154,3 +154,23 @@ func TestResubscribeWithErrorHandler(t *testing.T) {
 		t.Fatalf("unexpected subscription errors %v, want %v", subErrs, expectedSubErrs)
 	}
 }
+
+func TestResubscribeWithCompletedSubscription(t *testing.T) {
+	t.Parallel()
+
+	innerSubDone := make(chan struct{}, 1)
+	sub := ResubscribeErr(100*time.Millisecond, func(ctx context.Context, lastErr error) (Subscription, error) {
+		return NewSubscription(func(unsubscribed <-chan struct{}) error {
+			select {
+			case <-time.After(2 * time.Second):
```

Comment:
It's weird to have a timeout here. You can achieve the necessary synchronization event with another channel, like so

```
quitInner := make(chan struct{})
...
    select {
    case <-quitInner:
...

close(quitInner)
<-innerSubDone
sub.Unsubscribe()
```

Reply:
good suggestion. Made the change.